packages/web-components/fast-foundation/src/select/select.template.ts

```diff
@@ -0,0 +1,55 @@
+import { children, elements, html, ref, slotted } from "@microsoft/fast-element";
+import { endTemplate, startTemplate } from "../patterns/start-end";
+import { Select } from "./select";
+
+/**
+ * The template for the {@link @microsoft/fast-foundation#(Select:class)} component.
+ * @public
+ */
+export const SelectTemplate = html<Select>`
+    <template
+        class="${x => (x.open ? "open" : "")}"
+        ${children({ property: "options", filter: elements("fast-option") })}
+    >
+        <button
+            class="button"
+            ?disabled="${x => x.disabled}"
+            part="button"
+            ${ref("button")}
+        >
+            ${startTemplate}
+            <slot name="button-container" ${slotted("slottedButtonContainer")}>
+                <span class="selected-value" part="selected-value">
+                    <slot name="selected-value">
+                        ${x => x.displayValue}
+                    </slot>
+                </span>
+                <span class="indicator" part="indicator" ${ref("indicatorContainer")}>
+                    <slot name="indicator">
+                        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 7">
+                            <path
+                                d="M11.85.65c.2.2.2.5 0 .7L6.4 6.84a.55.55 0 01-.78 0L.14 1.35a.5.5 0 11.71-.7L6 5.8 11.15.65c.2-.2.5-.2.7 0z"
+                            />
+                        </svg>
+                    </slot>
+                </span>
+            </slot>
+            ${endTemplate}
+        </button>
+        <fast-listbox
```

Comment:
I'm not super excited by this implicit relationship to the `@microsoft/fast-components` package here, though I'm not 100% sure what to suggest.

Reply:
There is also a similar problem in data grid and presumably any component that creates other components at runtime.

Reply:
I'm thinking that when we declare a web component we should be able to set some properties as well, like what tags to use for generated options or grid rows.